nyxx_interactions/lib/src/Interactions.dart

```diff
@@ -78,91 +82,78 @@ class Interactions {
   }
 
   Future<void> _syncPermissions() async {
-    final commandPartition = _partition<SlashCommandBuilder>(
-        this._commandBuilders, (element) => element.guild == null);
+    final commandPartition = _partition<SlashCommandBuilder>(this._commandBuilders, (element) => element.guild == null);
     final globalCommands = commandPartition.first;
-    final groupedGuildCommands =
-        _groupSlashCommandBuilders(commandPartition.last);
-
-    final globalBody = globalCommands
-      .where((builder) => builder.permissions != null && builder.permissions!.isNotEmpty)
-      .map((builder) => {
-        "id": builder._id.toString(),
-        "permissions": [for (final permsBuilder in builder.permissions!) permsBuilder.build()]
-      })
-      .toList();
-
-    await this
-        ._client
-        .httpEndpoints
-        .sendRawRequest("/applications/${this._client.app.id}/commands/permissions", "PUT", body: globalBody);
+    final groupedGuildCommands = _groupSlashCommandBuilders(commandPartition.last);
+
+    final globalBody =
+        globalCommands.where((builder) => builder.permissions != null && builder.permissions!.isNotEmpty).toList();
+
+    if (globalBody.isNotEmpty) {
+      throw "Only guild commands may have permission's set. If you need custom permissions based on guild then you can send HTTP requests.";
```

Comment:
thats a bad practise. we should only throw Error or Exception instances. and in case of future you should return Future.error. You can return there `StateError` with same message